api/envoy/extensions/stat_sinks/open_telemetry/v3/open_telemetry.proto

```diff
@@ -20,8 +23,23 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // Stats configuration proto schema for ``envoy.stat_sinks.open_telemetry`` sink.
 // [#extension: envoy.stat_sinks.open_telemetry]
 
-// [#next-free-field: 8]
+// [#next-free-field: 9]
 message SinkConfig {
+  // Defines a custom conversion from a stat to a metric. This can be used to rename a
+  // stat, add static labels, and aggregate multiple stats into a single metric.
+  // If a stat's name matches ``stat_name_matcher``, it is converted to a metric with the
+  // given ``metric_name`` and``static_metric_labels`` added.
+  message CustomMetricConversion {
+    // The matcher to select stats for conversion.
+    type.matcher.v3.StringMatcher stat_name_matcher = 1;
```

Comment:
One thing I wonder if we should use a unified matcher for trie-based matching. If there are a lot of similar "prefix/suffix" patterns, it would make sense to organize them in a tree.

Reply:
Discussed with Kuat offline, we may still need regex for some cases like multiple stat label values are unknown at bootstrap time.

CC @markdroth